task/common/identifier.go

```diff
@@ -12,38 +12,46 @@ import (
 	"github.com/aohorodnyk/uid"
 )
 
-type Identifier string
+type Identifier struct {
+	name string
+	salt string
+}
 
 var ErrWrongIdentifier = errors.New("wrong identifier")
 
 const (
 	maximumLongLength = 50
 	shortLength       = 16
+	nameLength        = maximumLongLength-shortLength-uint32(len("tpi---"))
 )
 
 func ParseIdentifier(identifier string) (Identifier, error) {
 	re := regexp.MustCompile(`(?s)^tpi-([a-z0-9]+(?:[a-z0-9-]*[a-z0-9])?)-([a-z0-9]+)-([a-z0-9]+)$`)
 
 	if match := re.FindStringSubmatch(string(identifier)); len(match) > 0 && hash(match[1]+match[2], shortLength/2) == match[3] {
-		return Identifier(match[1]), nil
+		return Identifier{name: match[1], salt: match[2]}, nil
 	}
 
-	return Identifier(""), ErrWrongIdentifier
+	return Identifier{}, ErrWrongIdentifier
 }
 
-func NewIdentifier(identifier string) Identifier {
-	return Identifier(identifier)
+func NewDeterministicIdentifier(name string) Identifier {
```

Comment:
Public functions need a comment describing what they do.

Reply:
Thanks!

Not sure what you mean by "has no public API" - if a package has exported identifiers, it has what can be considered a public API.

Reply:
I mean that indeed, it has a “public API” in the sense that we can consume it ourselves, but the module is a Terraform provider and, as such:

1. Can't be imported by other projects.
2. Doesn't even appear in pkg.go.dev

Reply:
Surely we're not documenting things just for the sake of third-party developers, right?

The fact that we make tpi non-trivial to import into another go project by misdefining the module name has no bearing. And I'm not sure what you mean by saying it can't be imported into other projects - I whip up a working example of doing exactly that in a few minutes.

Reply:
> _Surely we're not documenting things just for the sake of third-party developers, right?_

For instance, DVC only has “docstrings” on those files/classes/functions that are meant to be imported by end users.  🤭 But yes, we should be the main audience for code documentation if we choose to.

Reply:
> _The fact that we make tpi non-trivial to import into another go project by misdefining the module name has no bearing._

It does, in the sense that whatever we document won't be useful for anybody else than developers and contributors.

Reply:
I guess that this conversation boils down to whether we want comments for ourselves or not. 🤔

Reply:
Resolving in favor of notion#LEO-improvements-a763b3a3308749e68ad9b53cc1438e8c